src/daemon/bitcoind/actions.rs

```diff
@@ -313,6 +314,19 @@ fn mark_confirmed_spends(
                 unvault_outpoint,
                 height
             );
+        } else if bitcoind.getmempoolentry(spend_txid).is_err() {
+            // At least, is this transaction still in mempool?
+            // If not, it's probably been RBF'd. Let's downgrade this to `unvaulting`,
```

Comment:
RBF is a special case, there are a lot of reasons for which a transaction could get evicted.